File: datasets_v2/finance/gnucash/swiftui/Fragment_scheduled_events_list.swift
import SwiftUI

struct Fragment_scheduled_events_list: View {
  var body: some View {
    VStack(alignment: .leading, spacing: 0) {
      ScrollView {
        LazyVStack(alignment: .leading, spacing: 4) {
          ForEach(1...10, id: \.self) { idx in
            VStack(alignment: .leading, spacing: 2) {
              Text("Item \(idx)")
              Text("Sub Item \(idx)")
            }
            Divider()
          }
        }
      }
      .frame(maxWidth: .infinity, maxHeight: .infinity)
      Text("No recurring transactions to display.").multilineTextAlignment(.center).frame(maxWidth: .infinity, alignment: .center).padding(.top, 8.0).padding(.bottom, 8.0).frame(maxWidth: .infinity, alignment: .center)
    }
    .frame(maxWidth: .infinity, maxHeight: .infinity)
  }
}

struct Fragment_scheduled_events_list_Previews: PreviewProvider {
  static var previews: some View {
    Fragment_scheduled_events_list()
  }
}

File: experiments/finance/gnucash/compose/fragment_scheduled_events_list.kt
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.paint
import androidx.compose.ui.graphics.Color
import androidx.compose.foundation.Image
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.sp
import androidx.compose.ui.unit.dp

@Composable
fun fragment_scheduled_events_list() {
  Column(modifier = Modifier.fillMaxWidth().fillMaxHeight()) {
    val listItems_android_list = (1..10).map { it }
    LazyColumn(modifier = Modifier.fillMaxWidth().fillMaxHeight()) {
      items(listItems_android_list) { idx ->
        Column(modifier = Modifier.fillMaxWidth().padding(vertical = 4.dp)) {
          Text(text = "Item $idx")
          Text(text = "Sub Item $idx")
        }
      }
    }
    Box(modifier = Modifier.fillMaxWidth(), contentAlignment = Alignment.TopCenter) {
    Text(text = "No recurring transactions to display.", modifier = Modifier.fillMaxWidth().fillMaxHeight().padding(top = 8.0.dp).padding(bottom = 8.0.dp), textAlign = TextAlign.Center)
    }
  }
}


@Preview(showBackground = true)
@Composable
fun fragment_scheduled_events_list_preview() {
    MyApplicationTheme {
        fragment_scheduled_events_list()
    }
}
